### ISSUE TITLE
Fluid benchmark & book validation

### ISSUE THREAD

**Issue by dzhwinter** [PaddlePaddle/Paddle#6208]
In the 0.11.0 version, we will release the book chapters written with `fluid`, there are some tasks need to be done. 
## Task Lists 1 : compare results with Paddle books V2

Need to validate these books can convergence to the approximate same result with books chapters.

- [ ] book.03 image classification **CPU** loss validation @jacquesqiao @qingqing01 @kuke 
- [ ] book.03 image classification **GPU** loss validation @jacquesqiao @qingqing01 @kuke 
     - [ ] ResNet
     - [ ] VGG

- [ ] book.04 word2vec **CPU** loss validation @peterzhang2029
- [ ] book.04 word2vec **GPU** loss validation @peterzhang2029

- [ ] book.05 recommendation systems **CPU** loss validation @typhoonzero 
- [ ] book.05 recommendation systems **GPU** loss validation @typhoonzero 

Need to note that we have three different implementation of understand_sentiment, only test the lstm one in this chapter. 
- [ ] book.06 understand_sentiment lstm **CPU** loss validation @ranqiu92
- [ ] book.06 understand_sentiment lstm **GPU** loss validation @ranqiu92

- [ ] book.07 label semantic roles  **CPU** loss validation @chengduoZH 
We do not have GPU version label semantic roles implementation.

- [ ] book.08 machine translation  **CPU** loss validation @jacquesqiao @ChunweiYan 
- [ ] book.08 machine translation **GPU** loss validation @jacquesqiao @ChunweiYan 

## Task Lists How to do
We have benchmark scripts and docker image. So these things should be done quickly and report a bug if you find any issue. (operator implement, convergence result). 
Because we are still finetuning the performance, so if you find any magnitude gap in performance, please file an issue without hesitation. 

scripts are put under this directory, please find the correct chapter name:
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/tests/book

old books docker image: 
**paddlepaddle/book:latest-gpu**

new books docker image:
**dzhwinter/benchmark:latest**

**Comment by dzhwinter** [comment]
## Task Lists 2 : compare results and performance with TensorFlow
We select some typical tasks, to validate our performance and results with TensorFlow.
Need to note that performance validation should be done in latest  Paddle with `Release` mode, set the same cuda/cudnn version. 


- [ ] book.02 mnist **GPU** performance validation with TensorFlow

- [ ] book.03 image classification **CPU** performance validation with TensorFlow
- [ ] book.03 image classification **GPU** performance validation with TensorFlow

- [ ] book.06 understand sentiment **CPU** performance validation with TensorFlow
- [ ] book.06 understand sentiment **GPU** performance validation with TensorFlow

## Task Lists 2 How to do
scripts:
https://github.com/dzhwinter/benchmark

both cpu tests can be done in docker image, but you need to install tensorflow-gpu by yourself:
dzhwinter/benchmark:latest

**Comment by dzhwinter** [comment]
recomend docker command. 
` nvidia-docker run -it -d --name mnist_gpu --security-opt seccomp=unconfined -v $PWD/benchmark:/benchmark -v /usr/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu dzhwinter/benchmark:latest  /bin/bash`

**Comment by chengduoZH** [comment]
`crf_decoding` is needed by `label senmantic roles`, but it does not support GPU on both v2 and fluid. 

**~~book.07 label senmantic roles GPU loss validation~~**

**Comment by reyoung** [comment]
This feature will be delayed to `0.11.1`

**Comment by dzhwinter** [comment]
We have released the Fluid 0.11.0 two days ago. Next stage will be introducing our new design to users. We need more solid metrics to compare with other frameworks,  find the potential flaw in Fluid. 
We maintain a repo to collect the common models, to make benchmark reproduce more easily. Need to notice that this repo and docker image will transfer to paddle in the future.


```bash
git clone https://github.com/dzhwinter/benchmark
docker pull dzhwinter/benchmark:latest
nvidia-docker run -it --name mnist_gpu --security-opt seccomp=unconfined -v $PWD/benchmark:/benchmark -v /usr/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu dzhwinter/benchmark:latest /bin/bash
```
https://github.com/dzhwinter/benchmark/blob/master/HowToDoBenchmark.md

Follow the guide step by step, then you will get the same environment.

To make the benchmark more convincing and general testing, there are some decisions need to make.

1. We select the most popular models- mnist, VGG-19,   Resnet, Stacked-lstm. Is there any more general benchmark models we need to include?

2. Do we need an op-wise compare framework to support our per-op precision checking? 
Namely, write an op test, one side is TensorFlow, one side is Fluid, help us debugging big models.


https://www.tensorflow.org/performance/benchmarks
https://github.com/pytorch/benchmark

TensorFlow and Pytorch focus on huge image related models, namely, CNN models. I think their test case is not general enough, but also can be a comparable sign.

**Comment by dzhwinter** [comment]
- [ ] seq2seq model / attention model

**Comment by pkuyym** [comment]
Popular dataset
http://yann.lecun.com/exdb/mnist
http://ufldl.stanford.edu/housenumbers
https://www.cs.toronto.edu/~kriz/cifar.html
